bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/menuitem/listener/LanguageItemSelectionListener.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright 2021-2022 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+
+package com.espressif.idf.ui.menuitem.listener;
+
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.widgets.MenuItem;
+import org.eclipse.ui.PlatformUI;
+
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.util.EclipseIniUtil;
+
+public class LanguageItemSelectionListener extends SelectionAdapter
+{
+	private static final String LANGUAGE_SWITCH = "-nl";
+	
+	EclipseIniUtil eclipseIniUtil;
+
+	public LanguageItemSelectionListener()
+	{
+		try
+		{
+			eclipseIniUtil = new EclipseIniUtil();
+		}
+		catch (Exception e)
+		{
+			IDFCorePlugin.getPlugin().getLog().error(e.getMessage());
+		}
+	}
+
+	@Override
+	public void widgetSelected(SelectionEvent selectionEvent)
+	{
+		MenuItem currentItem = ((MenuItem) selectionEvent.widget);
+		String code = currentItem.getText().split("-")[1].replaceAll("^[ \t]+|[ \t]+$", "");
+		if (eclipseIniUtil != null)
+		{
+			try
+			{
+				if (currentItem.getText().toLowerCase().contains("default"))
+				{
```

Comment:
While creating a menu item, use menuitem.setData(langcode) and while accessing you can get the lang code using menuitem.getData() and pass this key for switching

Reply:
Can use that but dont need to as we have now remove the separate listener class as you mentioned